wallet/script.cpp

```diff
@@ -1798,48 +1799,50 @@ bool ExtractDestinations(const CScript& scriptPubKey, txnouttype& typeRet,
     return true;
 }
 
-bool VerifyScript(const CScript& scriptSig, const CScript& scriptPubKey, const CTransaction& txTo,
-                  unsigned int nIn, bool fValidatePayToScriptHash, bool fStrictEncodings, int nHashType,
-                  ScriptError* serror)
+Result<void, ScriptError> VerifyScript(const CScript& scriptSig, const CScript& scriptPubKey,
+                                       const CTransaction& txTo, unsigned int nIn,
+                                       bool fValidatePayToScriptHash, bool fStrictEncodings,
+                                       int nHashType)
 {
-    set_error(serror, SCRIPT_ERR_UNKNOWN_ERROR);
 
     vector<vector<unsigned char>> stack, stackCopy;
-    if (!EvalScript(stack, scriptSig, txTo, nIn, fStrictEncodings, nHashType))
-        return false;
+
+    TRYV(EvalScript(stack, scriptSig, txTo, nIn, fStrictEncodings, nHashType));
+
     if (fValidatePayToScriptHash)
         stackCopy = stack;
-    if (!EvalScript(stack, scriptPubKey, txTo, nIn, fStrictEncodings, nHashType))
-        return false;
+
+    TRYV(EvalScript(stack, scriptPubKey, txTo, nIn, fStrictEncodings, nHashType));
+
     if (stack.empty())
-        return set_error(serror, SCRIPT_ERR_EVAL_FALSE);
+        return Err(SCRIPT_ERR_EVAL_FALSE);
 
     if (CastToBool(stack.back()) == false)
-        return set_error(serror, SCRIPT_ERR_EVAL_FALSE);
+        return Err(SCRIPT_ERR_EVAL_FALSE);
 
     // Additional validation for spend-to-script-hash transactions:
     if (scriptPubKey.IsPayToScriptHash()) {
         if (!scriptSig.IsPushOnly()) // scriptSig must be literals-only
-            return set_error(serror, SCRIPT_ERR_SIG_PUSHONLY);
+            return Err(SCRIPT_ERR_SIG_PUSHONLY);
 
         if (stackCopy.empty()) // Check to make sure that the stackCopy has elements too
-            return false;
+            return Err(SCRIPT_ERR_UNKNOWN_ERROR);
```

Comment:
I'm assuming SCRIPT_ERR_UNKNOWN_ERROR is the correct error to throw here, but I wanted to leave a comment to double check it since it is the only one so far where the previous error is different than the new error.

Reply:
This is kind of a corner case. There's some checks and processing before verifying signatures, and these errors can't be represented as a script error. So I just gave them these values.